### DISCUSSION TITLE
Limit fields like GraphQL with json-mask

### DISCUSSION THREAD

**Discussion by bestickley** [trpc/trpc#2592] [General]
Hi, I'm coming from GraphQL where I can easily limit the fields returned by my API. I wanted to know how I could do this with tRPC. I stumbled across [json-mask](https://www.npmjs.com/package/json-mask) which allows me to do just this. You can have your procedure accept a `fields` string to define the response. Hope that this helps others.

**Comment by KATT** [comment]
The inference on the output won't work though AFAICT

**Reply by bestickley** [reply]
Great point, @KATT. I guess you could cast the response on the client to `Pick<T, "fields" | "requested">`. I'm just starting out, so I'll post a better a solution if I find one.
